hazelcast/src/main/java/com/hazelcast/spi/discovery/DiscoveryStrategy.java

```diff
@@ -112,4 +115,27 @@ default PartitionGroupStrategy getPartitionGroupStrategy() {
      * @since 3.7
      */
     Map<String, String> discoverLocalMetadata();
+
+    /**
+     * Marks the passed {@link Address} as unhealthy, which prevents it from being offered as a
+     * viable endpoint in some {@link DiscoveryStrategy} implementations, usually prompting
+     * this endpoint to be periodically probed for liveliness. If not supported by the underlying
+     * implementation, then this call does nothing.
+     *
+     * @param address the address to mark as unhealthy
+     * @since 5.4
+     */
+    default void markEndpointAsUnhealthy(Address address) { }
```

Comment:
i would remove these 2 new methods from public api, if target is only using it for wan discovery. 
It can be made an implementation detail for wan discovery for now.

Reply:
Thanks @ahmetmircik, I've marked these methods as `@PrivateApi` in both `DiscoveryStrategy` and `DiscoveryService` for now 👍 https://github.com/hazelcast/hazelcast/pull/25364/commits/de3c107ceee89f0a859d916eed9269082f84c3ac